### DISCUSSION TITLE
什麼是Labview?

### DISCUSSION THREAD

**Discussion by lyien1021** [1101-GS4719-C/Python-Community-GS4719-C#1232] [General]
因為這學期上了Labview的課程，想說來簡單介紹一下Labview是什麼，還有說一下我目前的學習心得。

Labview是一種圖形語言，特別為儀器控制、資料擷取、數據量測、資料分析、系統模擬及結果呈現等目標而設計。可以把Labview想像成一個虛擬儀表，可以透過設計程式來自定義儀表板的功能。

以下是一個範例：
![image](https://user-images.githubusercontent.com/84632877/231994273-01f82432-b361-4e5d-9c9b-00a680b2a58f.jpeg)
輸出結果為：
![image](https://user-images.githubusercontent.com/84632877/231994755-b4176a4e-b06c-4d3c-bffd-72cbd00812da.jpeg)

此範例為模擬擲骰子的動作，輸入擲骰子的次數，則會輸出某個值（1～6）出現的次數。

目前的修課心得：
我覺得一開始學Labview的時候覺得很新鮮有趣，因為有很多可愛的圖形可以選🤣，然而隨著難度越來越難，我覺得自己已經有點跟不上老師的教學進度，常常沒辦法自己憑空寫出完整正確的程式碼，於是我最近去敦煌買了書，希望能有所幫助😢。總結來說，我覺得學習Labview必須擁有一定的邏輯能力加上反覆的練習增加熟悉度。

**Comment by curtis1017** [comment]
看完作者的分享後，我也對Labview產生了好奇心，未來我想要接觸。但又很怕自己沒有辦法雪好這一塊，希望作者未來還能分享更多的想法，然後我想要問，如果從完全沒有接觸到變成自己做的出來大概需要多少時間。

**Reply by lyien1021** [reply]
我覺得要看你想完成的難度為何，如果是真的要運用在生活中的專案的話，不僅要了解labview的操作，軟硬體的介面通訊也要有基本的了解，難度其實蠻高的。

**Reply by lyien1021** [reply]
還有如果你想嘗試的話，中央有提供serial number可以免費使用labview，可以去portal授權軟體服務那邊看看～

**Comment by DuxtinTai** [comment]
身為電機系同學，我來分享一下labview在電機系的應用:

自動化控制：LabVIEW的編程環境和圖形化編程方式，使其成為自動化控制系統的理想工具。電機工程師可以使用LabVIEW來設計和控制各種自動化設備，如機器人、機械手臂、自動化裝置等。

數據擷取與分析：LabVIEW可與各種數據擷取設備連接，例如儀器、傳感器、資料採集卡等，進行實時數據擷取和監控。此外，LabVIEW還支援各種數據分析工具和演算法，如傅立葉分析、小波分析、統計分析等，可幫助電機工程師進行數據分析和處理。

通信系統：LabVIEW擁有豐富的通信協定和通信控制工具，可用於開發各種通信系統，例如無線電系統、數據通信系統、語音通信系統等。此外，LabVIEW還支援各種通信協定和介面，如TCP/IP、USB、CAN等，可方便地進行通信控制和數據交換。

總之，LabVIEW是一個功能強大的圖形化編程工具，可廣泛應用於電機系的各個領域，包括電路設計、自動化控制、數據擷取與分析、通信系統等。
